File: src/scala/org/codehaus/testdox/intellij/actions/AutoscrollAction.scala
package org.codehaus.testdox.intellij.actions

import com.intellij.openapi.actionSystem.AnActionEvent

import org.codehaus.testdox.intellij.IconHelper
import org.codehaus.testdox.intellij.IconHelper._

class AutoscrollAction(private var selected: Boolean, useFromTestDoxToolWindow: Boolean)
    extends BaseToggleAction("Autoscroll To Source", "Toggle autoscrolling", getIcon(IconHelper.AUTOSCROLL_ICON), useFromTestDoxToolWindow) {

  def this() = this (false, false)

  def isSelected(event: AnActionEvent) = selected

  def setSelected(event: AnActionEvent, selected: Boolean) {
    this.selected = selected
    if (event != null) actionEvents.getTestDoxController(event).updateAutoscroll(selected)
  }
}

object AutoscrollAction {
  val ID = "TestDox.Autoscroll"
}

File: src/java/org/codehaus/testdox/intellij/actions/AutoscrollAction.java
package org.codehaus.testdox.intellij.actions;

import com.intellij.openapi.actionSystem.AnActionEvent;

import org.codehaus.testdox.intellij.IconHelper;

public class AutoscrollAction extends BaseToggleAction {

    public static final String ID = "TestDox.Autoscroll";

    private boolean selected = false;

    public AutoscrollAction() {
        this(false, false);
    }

    public AutoscrollAction(boolean selected, boolean useFromTestDoxToolWindow) {
        super("Autoscroll To Source", "Toggle autoscrolling", IconHelper.getIcon(IconHelper.AUTOSCROLL_ICON), useFromTestDoxToolWindow);
        this.selected = selected;
    }

    public boolean isSelected(AnActionEvent anActionEvent) {
        return selected;
    }

    public void setSelected(AnActionEvent event, boolean selected) {
        this.selected = selected;
        if (event != null) {
            actionEvents.getTestDoxController(event).updateAutoscroll(selected);
        }
    }
}
